pyqtgraph/graphicsItems/DateAxisItem.py

```diff
@@ -168,9 +168,15 @@ def tickValues(self, minVal, maxVal, minSpc):
             # reposition tick labels to local time coordinates
             ticks += self.utcOffset
             # remove any ticks that were present in higher levels
-            tick_list = [x for x in ticks.tolist() if x not in allTicks]
-            allTicks.extend(tick_list)
-            valueSpecs.append((spec.spacing, tick_list))
+            # we assume here that if the difference between a tick value and a previously seen tick value
+            # is less than min-spacing/100, then they are 'equal' and we can ignore the new tick.
+            close = np.any(
+                np.isclose(allTicks, ticks[:, np.newaxis], rtol=0, atol=minSpc * 0.01),
+                axis=-1,
+            )
+            ticks = ticks[~close]
+            allTicks = np.concatenate([allTicks, ticks])
+            valueSpecs.append((spec.spacing, ticks))
```

Comment:
To respect the description in the method docstring, `ticks` would need to be converted to a Python list here.